test/suite/test_compact08.py

```diff
@@ -0,0 +1,126 @@
+    table_uri='table:test_compact08'
+
+    delete_range_len = 10 * 1000
+    delete_ranges_count = 4
+    value_size = 1024 # The value should be small enough so that we don't create overflow pages.
+
+    def get_bg_compaction_running(self):
+        c = self.session.open_cursor('statistics:', None, 'statistics=(all)')
+        running = c[stat.conn.background_compact_running][2]
+        c.close()
+        return running
+
+    def get_bg_compaction_interrupted(self):
+        c = self.session.open_cursor('statistics:', None, 'statistics=(all)')
+        running = c[stat.conn.background_compact_interrupted][2]
+        c.close()
+        return running
+
+    # Create the table in a way that it creates a mostly empty file.
+    def test_compact08(self):
+
+        # FIXME-WT-11399
+        if self.runningHook('tiered'):
+            self.skipTest("this test does not yet work with tiered storage")
+
+        # Create the tables and populate them.
+        for i in range(0, self.num_tables):
+            uri = f'{self.table_uri}_{i}'
+            self.session.create(uri, self.create_params)
+            c = self.session.open_cursor(uri, None)
+            for k in range(self.table_numkv):
+                c[k] = ('%07d' % k) + '_' + 'abcd' * ((self.value_size // 4) - 2)
+            c.close()
+        self.session.checkpoint()
+
+        # Now let's delete a lot of data ranges. Create enough space so that compact runs in more
+        # than one iteration.
+        for i in range(0, self.num_tables):
+            uri = f'{self.table_uri}_{i}'
+            c = self.session.open_cursor(uri, None)
+            for r in range(self.delete_ranges_count):
+                start = r * self.table_numkv // self.delete_ranges_count
+                for i in range(self.delete_range_len):
+                    c.set_key(start + i)
+                    c.remove()
+            c.close()
+
+        num_interruption = 0
+        # Expect a message indication the interruption.
+        with self.expectedStderrPattern('background compact interrupted by application'):
+            # It is possible that the server was disabled between two tables which is not considered
+            # as an interruption, retry a few times if needed.
+            for i in range(0, 10):
+
+                # Background compaction should not be running yet.
+                self.assertEqual(self.get_bg_compaction_running(), 0)
+
+                # Start the background compaction server with a low threshold to make sure it starts
+                # working.
+                self.session.compact(None, 'background=true,free_space_target=1MB')
+
+                # Wait for the server to wake up.
+                running = self.get_bg_compaction_running()
+                while not running:
+                    running = self.get_bg_compaction_running()
+                self.assertEqual(running, 1)
+
+                # Disable the server which may interrupt compaction.
```

Comment:
I am not sure that in slow machines where the background compact server can get a chance to start compaction on a table before we disable it. How about adding 1 second sleep here to increase the probability of bg compact starts the compaction?

Reply:
Good idea, see 2fa8221474751f22a9d2bca27d1d0e86820ec755 where I am retrieving the stat `btree_compact_pages_rewritten` to make sure we are doing work.